hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGrantKycToAccountHandler.java

```diff
@@ -70,7 +72,22 @@ public void preHandle(@NonNull final PreHandleContext context, @NonNull final Re
      *
      * @throws NullPointerException if one of the arguments is {@code null}
      */
-    public void handle() {
-        throw new UnsupportedOperationException("Not implemented");
+    public void handle(@NonNull final TransactionBody txnBody, @NonNull WritableTokenRelationStore tokenRelStore) {
+        requireNonNull(txnBody);
+        requireNonNull(tokenRelStore);
+
+        final var op = txnBody.tokenGrantKycOrThrow();
+
+        /* --- Do the business logic --- */
+        final var targetTokenId = op.tokenOrThrow();
+        final var targetAccountId = op.accountOrThrow();
+        final var tokenRelation =
+                tokenRelStore.getForModify(targetTokenId.tokenNum(), targetAccountId.accountNumOrThrow());
+
+        /* --- Persist the updated models --- */
+        final var tokenRelBuilder = tokenRelation.orElseThrow().copyBuilder();
+        tokenRelBuilder.kycGranted(true);
+        tokenRelStore.put(tokenRelBuilder.build());
+        tokenRelStore.commit();
```

Comment:
We should probably move this to `TransactionDispatcher` dispatch method. To keep all `commit` in one place.